fvtest/compilertest/tests/MetaDataTest.cpp

```diff
@@ -0,0 +1,159 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "avl_api.h"
+#include "env/TRMemory.hpp"
+#include "runtime/CodeMetaDataManager.hpp"
+#include "runtime/CodeMetaDataPOD.hpp"
+#include <vector>
+
+
+/**
+ * The design of the CodeMetaData manager is such that it expects each code
+ * cache to be added to the CodeMetaDataManager. In order to avoid that
+ * coupling in this test, the below class allows allocating 'fake' homes for
+ * tree elements
+ */
+
+class TestEnhancedCodeMetaDataManager : public TR::CodeMetaDataManager {
+   public:
+   TestEnhancedCodeMetaDataManager() : TR::CodeMetaDataManager() {}
+
+   /**
+    * Reset the singleton on destruction.
+    */
+   ~TestEnhancedCodeMetaDataManager()
+      {
+      _codeMetaDataManager = NULL;
+      }
+
+   TR::MetaDataHashTable* addFalseCacheInfo(uintptr_t base, uintptr_t top) {
+
+      TR::MetaDataHashTable *newTable = allocateCodeMetaDataHash(base,top);
+
+      if (newTable)
+         {
+         avl_insert(_metaDataAVL, (J9AVLTreeNode *) newTable);
+         }
+
+      return newTable;
+   }
+};
+
+
+TEST(MetaDataTest, Construction) {
+   TR::CodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+}
+
+
+TEST(MetaDataTest, ExactInsertion) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+
+   TR::MethodMetaDataPOD data;
+   data.startPC = 10;
+   data.endPC   = 100;
+   data.name    = "test";
+
+   ASSERT_TRUE(manager.insertMetaData(&data));
+   ASSERT_TRUE(manager.containsMetaData(&data));
+
+   auto* metadata = manager.findMetaDataForPC(10);
```

Comment:
I think
```c++
auto* metadata = manager.findMetaDataForPC(data.startPC);
```
is clearer.

Also is there a reason for using `auto*` instead of just `auto`?